elfi/methods/inference/bolfire.py

```diff
@@ -0,0 +1,447 @@
+        target_model: GPyRegression, optional
+            A surrogate model to be used.
+        acquisition_method: Acquisition, optional
+            Method of acquiring evidence points. Default LCBSC.
+
+        """
+        # Resolve model and initialize
+        model = self._resolve_model(model)
+        super(BOLFIRE, self).__init__(model, output_names=None, *args, **kwargs)
+
+        # Initialize attributes
+        self.n_training_data = self._resolve_n_training_data(n_training_data)
+        self.summary_names = self._get_summary_names(self.model)
+        self.marginal = self._resolve_marginal(marginal, seed_marginal)
+        self.classifier = self._resolve_classifier(classifier)
+        self.observed = self._get_observed_summary_values(self.model, self.summary_names)
+        self.prior = ModelPrior(self.model)
+
+        # TODO: write resolvers for the attributes below
+        self.bounds = bounds
+        self.acq_noise_var = acq_noise_var
+        self.exploration_rate = exploration_rate
+        self.update_interval = update_interval
+
+        # Initialize GP regression
+        self.target_model = self._resolve_target_model(target_model)
+
+        # Initialize BO
+        self.n_initial_evidence = self._resolve_n_initial_evidence(n_initial_evidence)
+        self.acquisition_method = self._resolve_acquisition_method(acquisition_method)
+
+        # Initialize state dictionary
+        self.state['n_evidence'] = 0
+        self.state['last_GP_update'] = self.n_initial_evidence
+
+        # Initialize classifier attributes list
+        self.classifier_attributes = []
+
+    @property
+    def n_evidence(self):
+        """Return the number of acquired evidence points."""
+        return self.state['n_evidence']
+
+    def set_objective(self, n_evidence):
+        """Set an objective for inference. You can continue BO by giving a larger n_evidence.
+
+        Parameters
+        ----------
+        n_evidence: int
+            Number of total evidence for the GP fitting. This includes any initial evidence.
+
+        """
+        if n_evidence < self.n_evidence:
+            logger.warning('Requesting less evidence than there already exists.')
+        self.objective['n_sim'] = n_evidence
+
+    def extract_result(self):
+        """Extract the results from the current state."""
+        return BOLFIREPosterior(self.parameter_names,
+                                self.target_model,
+                                self.prior,
+                                self.classifier_attributes)
+
+    def update(self, batch, batch_index):
+        """Update the GP regression model of the target node with a new batch.
+
+        Parameters
+        ----------
+        batch : dict
+            dict with `self.outputs` as keys and the corresponding outputs for the batch
+            as values
+        batch_index : int
+            Index of batch.
+
+        """
+        super(BOLFIRE, self).update(batch, batch_index)
```

Comment:
this updates num batches and simulations as:
```
self.state['n_batches'] += 1
self.state['n_sim'] += self.batch_size
```
but in this method `batch_size` indicates num parameter combinations tested, not num simulations. so we need:
```
self.state['n_batches'] += 1
self.state['n_sim'] += self.batch_size * self.n_training_data
```